From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gellert Hegyi <gellert.hegyi@around.co>
Date: Wed, 20 Oct 2021 10:50:14 +0200
Subject: exposes allowFragCoord in SkRuntimeEffect

This patch exposes allowFragCoord in SkRuntimeEffect so we can use sk_FragCoord in our video shader filter code.

diff --git a/include/effects/SkRuntimeEffect.h b/include/effects/SkRuntimeEffect.h
index b3f21b1d58e628c18146ee4196a5b6f5aa48ba2a..8134cb75628570ae501b20aaf17241b56380ec88 100644
--- a/include/effects/SkRuntimeEffect.h
+++ b/include/effects/SkRuntimeEffect.h
@@ -97,6 +97,13 @@ public:
         // run the inliner directly, but they still get an inlining pass once they are painted.)
         bool forceNoInline = false;
 
+        // Similarly: Public SkSL does not allow access to sk_FragCoord. The semantics of that
+        // variable are confusing, and expose clients to implementation details of saveLayer and
+        // image filters.
+        // AROUND: We expose 'allowFragCoord' so we can use sk_FragCoord in our
+        // video filter shader code.
+        bool allowFragCoord = false;
+
     private:
         friend class SkRuntimeEffect;
         friend class SkRuntimeEffectPriv;
@@ -106,11 +113,6 @@ public:
         // still largely ES3-unaware and can still fail or crash if post-ES2 features are used.
         // This is only intended for use by tests and certain internally created effects.
         bool enforceES2Restrictions = true;
-
-        // Similarly: Public SkSL does not allow access to sk_FragCoord. The semantics of that
-        // variable are confusing, and expose clients to implementation details of saveLayer and
-        // image filters.
-        bool allowFragCoord = false;
     };
 
     // If the effect is compiled successfully, `effect` will be non-null.
